runtime/runtime/src/verifier.rs

```diff
@@ -144,20 +150,48 @@ pub fn verify_and_charge_transaction(
         )?;
     let transaction = &signed_transaction.transaction;
     let signer_id = &transaction.signer_id;
+    let public_key = &transaction.public_key;
 
     let mut signer = match get_account(state_update, signer_id)? {
         Some(signer) => signer,
         None => {
             return Err(InvalidTxError::SignerDoesNotExist { signer_id: signer_id.clone() }.into());
         }
     };
-    let mut access_key = match get_access_key(state_update, signer_id, &transaction.public_key)? {
+
+    #[cfg(feature = "protocol_feature_eth_implicit")]
+    let mut access_key = get_access_key(state_update, signer_id, public_key)?;
+    #[cfg(not(feature = "protocol_feature_eth_implicit"))]
+    let access_key = get_access_key(state_update, signer_id, public_key)?;
+
+    #[cfg(feature = "protocol_feature_eth_implicit")]
+    if checked_feature!("stable", EthImplicit, current_protocol_version) {
+        // Access key is missing, but in the case of a transaction from ETH-implicit account
+        // there is a possibility that full access key will be created now.
+        if access_key.is_none() && signer_id.get_account_type() == AccountType::EthImplicitAccount {
+            if derive_account_id_from_public_key(public_key) == *signer_id {
+                // TODO What about increasing storage usage for that account, because we added access key?
```

Comment:
We add access key here. That increases storage for that account. Would it be correctly handled later?

Reply:
See my comment above about potentially not needing this logic. But either way, I think duplicating the code is not ideal and there could be a function that encapsulates this in both `action.rs` and `verifier.rs`.